backend/src/middleware/validation/validationMiddleware.js

```diff
@@ -115,12 +116,31 @@ const validateReview = async (resolve, root, args, context, info) => {
   return resolve(root, args, context, info)
 }
 
+export const validateNotifyUsers = async (label, reason) => {
+  const reasonsAllowed = ['mentioned_in_post', 'mentioned_in_comment', 'commented_on_post']
+  if (!reasonsAllowed.includes(reason)) throw new Error('Notification reason is not allowed!')
+  if (
+    (label === 'Post' && reason !== 'mentioned_in_post') ||
+    (label === 'Comment' && !['mentioned_in_comment', 'commented_on_post'].includes(reason))
+  ) {
+    throw new Error('Notification does not fit the reason!')
+  }
+}
+
+const validateUpdateUser = async (resolve, root, params, context, info) => {
+  const { name } = params
+  if (typeof name === 'string' && name.trim().length < USERNAME_MIN_LENGTH)
+    throw new UserInputError(`Username must be at least ${USERNAME_MIN_LENGTH} character long!`)
+  return resolve(root, params, context, info)
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-12-12T23:53:10Z" title="Friday, December 13th 2019, 12:53:10 am +01:00">Dec 13, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -115,12 +116,32 @@ const validateReview = async (resolve, root, args, context, info) => {
   return resolve(root, args, context, info)
 }
 
+export const validateNotifyUsers = async (label, reason) => {
+  const reasonsAllowed = ['mentioned_in_post', 'mentioned_in_comment', 'commented_on_post']
+  if (!reasonsAllowed.includes(reason)) throw new Error('Notification reason is not allowed!')
+  if (
+    (label === 'Post' && reason !== 'mentioned_in_post') ||
+    (label === 'Comment' && !['mentioned_in_comment', 'commented_on_post'].includes(reason))
+  ) {
+    throw new Error('Notification does not fit the reason!')
+  }
+}
+
+const validateUpdateUser = async (resolve, root, params, context, info) => {
+  const { name } = params
+  if (typeof name === 'string' && name.trim().length > USERNAME_MIN_LENGTH)
+    return resolve(root, params, context, info)
+  if (typeof name !== 'string' || name.trim().length < USERNAME_MIN_LENGTH)
```


I would say just remove the if check, no? Is there a reason for it?